### DISCUSSION TITLE
Custom screen breakpoint overrides default bigger breakpoint

### DISCUSSION THREAD

**Discussion by ViktorShev** [tailwindlabs/tailwindcss#12986] [Help] [answered]
Tailwind version: `"tailwindcss": "^3.4.1"`

I have the following in my `tailwind.config.ts`:

Note: screens that start with a double underscore are deprecated and not used in new code, they were introduced by the previous development team before me but i had to keep them for backwards compatibility. The ones added by me (and the ones i'm experiencing issues with are the ones _without_ a double underscore.

```
const theme = {
  screens: {
    __md: { min: '769px', max: '992px' },
    __MidScreen: { min: '767px', max: '991px' },
    __lg: { min: '993px', max: '1200px' },
    __xl: { min: '1201px' },
    __macOS: { min: '992px', max: '1399px' },
    __smallScreen: { max: '820px' },
    __MobileView: { max: '767px' },
    __MobileVieww: { max: '769px' },
    __TabletViewww: { min: '760px', max: '806px' },
    '__container-screen': { min: '992px', max: '1199px' },
    __mediumview: { max: '992px' },
    __largerView: { min: '1315px' },
    __smallerView: { min: '100px', max: '992px' },
    __ipadScreen: { min: '768px', max: '1024px' },
    __largeScreen: { min: '993px' },
    __forDesktop: { min: '768px' },
    __max1280: { max: '1280px' },
    __above767: { min: '768px' },
    __largeScreen2: { min: '1200px' },
    __xsMin: { min: '601px' },
    __xsToipad: { min: '601px', max: '1280px' },
    __xsTotab: { min: '601px', max: '767px' },
    __xsTotabLand: { min: '601px', max: '991px' },
    __smallMobile: { max: '413px' },
    __smallMobile2: { min: '350px', max: '413px' },
    __sm: { min: '601px', max: '768px' },
    __xs: { max: '600px' },
    xs: '375px',
    mobile: '390px',
    tablet: '601px',
    ...defaultScreens,
    desktop: '1200px',
    lg: defaultScreens.lg,
    xl: defaultScreens.xl,
    '2xl': defaultScreens['2xl'],
    'max-xs': { max: '374px' },
    'max-mobile': { max: '389px' },
    'max-tablet': { max: '600px' },
    'max-sm': { max: '639px' },
    'max-md': { max: '767px' },
    'max-lg': { max: '1023px' },
    'max-desktop': { max: '1199px' },
    'max-xl': { max: '1279px' },
    'max-2xl': { max: '1535px' },
    'from-md-to-desktop': { min: '768px', max: '1199px' },
    'from-sm-to-md': { min: '640px', max: '767px' }
  },
  extend: {
    ...
    },
```

As you can see I am **overriding** the screens key on the `theme` config due to me adding smaller breakpoints than the defaults, and as you can see they are all sorted in ascending order, but for example, my `tablet` breakpoint is overriding the default `md` breakpoint: 

![image](https://github.com/tailwindlabs/tailwindcss/assets/84634751/448c5eac-ebe2-47da-aa8c-bf9f76c6b34f)

Then when we reach the `desktop` breakpoint, it's overriding the `tablet` one correctly (as expected):

![image](https://github.com/tailwindlabs/tailwindcss/assets/84634751/d1acb443-f01d-4401-8c24-edef1fb22e6a)

I've been banging my head for way too long trying to find the issue and it seems there's barely any other people that are experiencing this issue online as well, the only comment I found which seems to be dealing with the same issue is this: https://github.com/tailwindlabs/tailwindcss/issues/1319#issuecomment-739888874, but the issue was closed and their question was never answered.

This is causing me to use `!` every time I want to utilize my custom breakpoints and the default ones at the same time, which is not great.

Thanks.

**Comment by wongjn** [comment, marked as answer]
Seems to be in the correct order in [Tailwind Play](https://play.tailwindcss.com/f6bsM5Jm31) (open the *Generated CSS* panel) which would suggest some other misconfiguration/misuse. Consider providing a project that reproduces the unexpected behavior.

**Reply by ViktorShev** [reply]
You're right, I'm using a package `@material-tailwind/react": "2.1.1"`, which exports a wrapper over your tailwind config `withMT`, if I exported the tailwind config wrapped by this function call it would produce the following screens config, putting `md` before my `tablet` breakpoint:

![Screenshot_20240221_160434](https://github.com/tailwindlabs/tailwindcss/assets/84634751/3d294732-0852-4230-82a0-e36d923025d5)

Thanks.
